src/domain/entities/Notification.ts

```diff
@@ -1,13 +1,37 @@
 import { NamedRef } from "./Ref";
+import { Struct } from "./generic/Struct";
+import { User } from "./User";
 
-export type Notification = {
+export type NotificationAttrs = {
     id: string;
     content: string;
     recipients: NotificationRecipients;
     readBy: UserReadNotification[];
     createdAt: Date;
 };
 
+export class Notification extends Struct<NotificationAttrs>() {
+    markAsRead(user: User): Notification {
+        if (this.isRead(user)) {
+            return this;
+        } else {
+            return this._update({
+                readBy: [
+                    ...this.readBy,
+                    {
+                        id: user.id,
+                        name: user.name,
+                        date: new Date(),
+                    },
+                ],
+            });
+        }
+    }
+    isRead(user: User): boolean {
```

Comment:
blank line between methods. Also, maybe it's more declarative: `isReadBy(user: User)`?